src/typescript/decorators.d.ts

```diff
@@ -0,0 +1,54 @@
+/// <reference path="../../node_modules/reflect-metadata/Reflect.d.ts" />
```

Comment:
Ok, and its use here is a requirement of interfacing with TypeScript, right? I'm completely opposed to that 'spec'; I believe it's very poorly thought out. But if it's *required* by TypeScript, then it makes sense to include in a PR for interfacing with it.